tasks/eva_3091/detect_cross_species_remapping.py

```diff
@@ -0,0 +1,138 @@
+                taxonomy_to_current_assembly[taxonomy] = assembly
+        assemblies_to_full_clustering_dates = defaultdict(list)
+        assemblies_to_species_clustering_dates = defaultdict(list)
+
+        # Get the assemblies where clustering for whole species was performed
+        query = 'select taxonomy, assembly_accession, clustering_start from eva_progress_tracker.clustering_release_tracker where should_be_clustered=true;'
+        for taxonomy, assembly, clustering_start in get_all_results_for_query(pg_conn, query):
+            assemblies_to_full_clustering_dates[assembly].append(clustering_start)
+
+        # Get the assemblies where clustering for new studies was performed
+        query = 'select taxonomy, assembly_accession, remapping_start, study_accessions from eva_progress_tracker.remapping_tracker where study_accessions is not null;'
+        for taxonomy, assembly, remapping_start, study_accessions in get_all_results_for_query(pg_conn, query):
+            assemblies_to_species_clustering_dates[(taxonomy, assembly)].append(remapping_start)
+
+    for taxonomy in taxonomy_to_current_assembly:
+        for assembly in taxonomy_to_assemblies[taxonomy]:
+            taxonomy_from_assembly = cached_get_taxonomy_from(assembly)
+            # No difference between the taxonomy and the assembly
+            if taxonomy == taxonomy_from_assembly:
+                continue
+            # taxonomy are different but the target assembly is the same as the current one for this
+            if assembly == taxonomy_to_current_assembly.get(taxonomy_from_assembly):
+                # logger.info(f'For {taxonomy} target is {taxonomy_to_assembly[taxonomy]} which comes from {taxonomy_from_assembly} with same target {taxonomy_to_assembly.get(taxonomy_from_assembly)}')
+                continue
+            # taxonomy are different and assembly is different from the current target
+            # but the target's taxonomy has itself no remapping target
+            if taxonomy_to_current_assembly.get(taxonomy_from_assembly) is None:
+                # logger.info(f'For {taxonomy} target is {taxonomy_to_assembly[taxonomy]} which comes from {taxonomy_from_assembly} with same target {taxonomy_to_assembly.get(taxonomy_from_assembly)}')
+                continue
+
+            # Check that if a clustering was ever performed on this assembly
+            if assembly not in assemblies_to_full_clustering_dates and \
+                    (taxonomy, assembly) not in assemblies_to_species_clustering_dates:
+                continue
+            dates_for_orig_assembly = ','.join([d.date().isoformat() if d else 'Unknown' for d in assemblies_to_full_clustering_dates.get(assembly, [])])
+            dates_for_orig_assembly += ','.join([d.date().isoformat() if d else 'Unknown' for d in assemblies_to_species_clustering_dates.get((taxonomy, assembly), [])])
+
+            # Check that if a clustering was ever performed on the other assembly
```

Comment:
Just to check my understanding: if we have `asm_1a` associated with `tax_1` but also containing variants from `tax_2`, and `tax_1`'s current supported assembly is `asm_1b`, then `asm_1a` is what we're calling the "original assembly" and `asm_1b` is the "other assembly".  Is that right?

Reply:
Yes that's correct, I should have a paragraph on nomenclature somewhere in the script/ticket